### ISSUE TITLE
AdVault export problem

### ISSUE THREAD

**Issue by TirOFlanc** [dhowe/AdNauseam#2503]
### I tried to reproduce the issue when...

- [X] AdNauseam is the only extension
- [X] AdNauseam uses default lists and settings
- [X] Using a new, unmodified browser profile

### Description

When I click on the "Export" button in the "AdVault" window, nothing happens.

I would like to know if there is a way to export the AdVault manually?

### Configuration

I'm running AdNauseam version 3.18.0, Brave 1.47.186, Windows 7 x64.

**Comment by dhowe** [comment]
There is no dialog/confirm after the ads are exported -- my guess (pls confirm) is that you have many json files in your downloads folder now...

**Comment by mneunomne** [comment]
I tested it using Brave and Adnauseam v3.18.0 as you mentioned. Could not reproduce the issue. Some possible issues might be occuring:
- too many ads are being selected? (need to test)
- Windows OS specific problem

**Comment by TirOFlanc** [comment]
This is when I click on "Export" (I don't know if there's an answer there).

![image](https://github.com/dhowe/AdNauseam/assets/56934692/0025dac5-0726-46ec-8219-b9d9ecf0d458)

**Comment by TirOFlanc** [comment]
I'm not sure I know how to read and understand what that means, but I have a feeling it's reaching the timeout?

![image](https://github.com/dhowe/AdNauseam/assets/56934692/bd0f8159-f42a-4dfd-b11a-be28a3620e28)

I'll try your code and report back to you, thanks.

**Comment by TirOFlanc** [comment]
Here's the result (right-click "Save As")

[dkoaabhijcomjinndlgbmfnmnjnmdeeb-1705329448117.log](https://github.com/dhowe/AdNauseam/files/13940042/dkoaabhijcomjinndlgbmfnmnjnmdeeb-1705329448117.log)

**Comment by mneunomne** [comment]
Then try this, see if you can download the file, it seems that the data was indeed being shown there in the console:

```js
vAPI.messaging.send('dashboard', {
        what: 'backupUserData',
    }).then((data) => {
    console.log("done", data)

    var jsonData = JSON.stringify(data.userData, null, '  ')
    let filename = "adnauseam-download-test";
    const url = URL.createObjectURL(new Blob([jsonData], { type: "text/plain" }));

    vAPI.download({
      'url': url,
      'filename': filename
    });
}).catch((err) => {
  console.error("error", err)  
});
```

**Comment by TirOFlanc** [comment]
Oui, cette fois j'obtiens bien la fenêtre d'enregistrement.

![image](https://github.com/dhowe/AdNauseam/assets/56934692/ca0df0ef-18d1-475c-aaff-9240f9792a71)

**Comment by mneunomne** [comment]
Ah good! yes. But I just named it wrongly. change it to:

```
AdNauseam_AdsNauseam_15.10.202416.16.16.json
```

Check if the file contains a json file with the list of ads. Maybe you can try to test it in another browser profile, or different browser, to see if the import will work or not.

**Comment by TirOFlanc** [comment]
Ok well, I'll test the import before overwriting everything just to be sure.

If there's a problem, I'll get back to you. Otherwise, consider the problem solved.

Were you able to figure out what was blocking the export?

**Comment by mneunomne** [comment]
No, but need to check what is happening. If you don't mind sharing your exported ads file so I could potentially reproduce this, but I understand if you don't want to for privacy reasons. Send me to `alberto.harres at gmail.com` if you have no problem with that.

**Comment by TirOFlanc** [comment]
I haven't tested the import yet, but if you compare it with an old backup, you'll see that it also exports the extension settings.

![image](https://github.com/dhowe/AdNauseam/assets/56934692/4c67735a-5978-4a2f-a63d-64de766f9ecd)

**Comment by TirOFlanc** [comment]
Hi, there,

It's me again. I wanted to redo a backup today as previously stated but I couldn't.

Other problems occurred. Here are some screenshots showing them:

![advault1](https://github.com/dhowe/AdNauseam/assets/56934692/1f0a7135-1996-4767-9f36-85f84e068bcf)
![advault2](https://github.com/dhowe/AdNauseam/assets/56934692/d06f64c3-5cdf-4876-867b-d5506c322df5)
![advault3](https://github.com/dhowe/AdNauseam/assets/56934692/0928b94d-bb8e-48e1-94e7-279f8fc63eaa)

I'll email you the file I created last time.
Thank you.

**Comment by TirOFlanc** [comment]
Hi there,

Yesterday I imported my Brave profile (%AppData%\local\BraveSoftware\...) into a freshly downloaded Brave portable (v1.47.186) (latest version supporting Windows 7), on a Windows 7 laptop too, with AdNauseam v3.18 then v3.20.

I encountered the same problem, namely that the ads don't display in the Vault as seen in the message just above.

I tried again, this time importing the file you'd managed to get me to generate via your JS code (which I emailed you), same problem.

I tried with an old backup of my AdVault that I had saved from October 10, 2023 and the export works without a hitch.

![image](https://github.com/dhowe/AdNauseam/assets/56934692/4946eef8-c644-4433-b00d-c10c0eae0531)

Thank you.

**Comment by mneunomne** [comment]
hello @TirOFlanc, I'm trying to fix the data that you have sent me by email, I'll ket you know when I manage to fix it. It seems that there was some issue with the control characters that is not making possible to re-import. I'm trying to clean the file so you can re-import once again.

**Comment by mneunomne** [comment]
@TirOFlanc I managed to fix the exported JSON file. I will send you back the corrected file by email right away.

**Comment by TirOFlanc** [comment]
Hi there,

Thanks for the reply.

I tried importing into a new brave installation but it didn't work.

![Capture1](https://github.com/dhowe/AdNauseam/assets/56934692/33ace31c-446c-4e0b-b9c5-1661a06ae627)
![Capture2](https://github.com/dhowe/AdNauseam/assets/56934692/2d18a95d-b2dd-4af6-b850-316975c13a69)

Would it be possible to ensure that, when the AdVault is exported, unauthorized characters (those not in a predefined list, for example) are simply deleted to avoid any problems of this kind?

Thank you for your help.

**Comment by dhowe** [comment]
> [@TirOFlanc](https://github.com/TirOFlanc) is there any errors in the console?
> 
> All the ads will be exported regardless of the selected area actually, so maybe thats indeed the problem. 33k ads its indeed a lot.

I've just tested import and export with 30,000 ads on current Chromium/AdNauseam with no problems. @mneunomne have you verified whether special characters may be causing the issue ?

**Comment by worldoftoz** [comment]
![Image](https://github.com/user-attachments/assets/16ab6bd6-d342-4462-b269-9d3bdd33862b)

**Comment by TirOFlanc** [comment]
Personally, I have given up on all that, the collection of ads and AdNauseam. Not that I don't like this extension anymore, but under Brave, there are no ads and I no longer have to deal with these problems.

**Comment by mneunomne** [comment]
> I've tested import and export with 30,000 ads on the latest Chromium/AdNauseam without issues. [@mneunomne](https://github.com/mneunomne), have you checked whether special characters could be causing the problem?

@dhowe, I've tried multiple times to reproduce this but haven’t been able to. Since this issue prevents the user from exporting their ads, it becomes even harder to diagnose.

@TirOFlanc, I’m sorry you're dealing with this—it’s frustrating to lose access to so much collected data.

I wonder if we can debug the JSON to identify where the corruption occurred and understand what went wrong. It’s also possible that something in the browser profile led to the data corruption.
